Tasks/Common/VstsAzureHelpers_/InitializeFunctions.ps1

```diff
@@ -41,6 +41,11 @@ function Initialize-AzureSubscription {
 
     #Set UserAgent for Azure Calls
     Set-UserAgent
+
+    $environmentName = "AzureCloud"
+    if( $Endpoint.Data.Environment ) {
```

Comment:
this does not validate if it contains an empty string with space. Please check if you want to trim it before it gets used